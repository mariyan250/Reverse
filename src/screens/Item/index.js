// React and style
import React, { useEffect } from 'react';
import './index.scss';

// Redux
import { connect } from 'react-redux';
import { getPost } from 'store/actions/post';

// Image Gallery
import ImageGallery from 'react-image-gallery';

function Item({ getPost, post, match }) {
  useEffect(() => {
    getPost(match.params.id);
  }, [getPost, post, match]);

  const images = [
    {
      original: 'https://picsum.photos/id/1018/1000/600/',
      thumbnail: 'https://picsum.photos/id/1018/250/150/',
    },
    {
      original: 'https://picsum.photos/id/1015/1000/600/',
      thumbnail: 'https://picsum.photos/id/1015/250/150/',
    },
    {
      original: 'https://picsum.photos/id/1019/1000/600/',
      thumbnail: 'https://picsum.photos/id/1019/250/150/',
    },
    {
      original: 'https://picsum.photos/id/1018/1000/600/',
      thumbnail: 'https://picsum.photos/id/1018/250/150/',
    },
    {
      original: 'https://picsum.photos/id/1015/1000/600/',
      thumbnail: 'https://picsum.photos/id/1015/250/150/',
    },
    {
      original: 'https://picsum.photos/id/1019/1000/600/',
      thumbnail: 'https://picsum.photos/id/1019/250/150/',
    },
  ];

  return (
    <section className="item">
      <ImageGallery
        items={images}
        showPlayButton={false}
        showThumbnails={false}
        showBullets={true}
        showFullscreenButton={false}
      />

      <article className="item-description">
        <h1>{post.name}</h1>
      </article>
    </section>
  );
}

const mapStateToProps = (state) => ({ post: state.posts.item });

export default connect(mapStateToProps, { getPost })(Item);
