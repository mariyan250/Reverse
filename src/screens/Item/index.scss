@import '~react-image-gallery/styles/scss/image-gallery.scss';
@import '../../index.scss';

.item-screen {
  display: grid;

  @include media-breakpoint-lg-up {
    grid-template-columns: to-rem(600px) 1fr;
  }

  @include media-breakpoint-xl-up {
    grid-template-columns: to-rem(700px) 1fr;
  }
}
