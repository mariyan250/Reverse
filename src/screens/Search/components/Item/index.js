// React and Style
import React from 'react';
import './index.scss';

// Router
import { Link } from 'react-router-dom';

// Translation
import { useTranslation } from 'react-i18next';

function Item(props) {
  const { t } = useTranslation();

  return (
    <li className="item app-bg-white box-shadow" {...props}>
      <Link to={`/item/${props.id}`} className="app-text-secondary">
        <section className="img-container d-flex overflow-hidden position-relative">
          <img
            src={props.images[0]}
            alt={t('search_screen.item.img.alt')}
            className="img-fluid position-absolute"
          />
        </section>

        <section className="p-3">
          <h2 className="my-3 text-center">{props.name}</h2>
          <p className="mt-4 mb-2 font-weight-light">
            <i className="fas fa-map-marker-alt mr-2" />
            {props.location}
          </p>
          <article className="d-flex justify-content-between align-items-end">
            <p className="m-0">
              <i className="fas fa-clock mr-2" />
              {props.date}
            </p>
            <h4>{props.price} $</h4>
          </article>
        </section>
      </Link>
    </li>
  );
}

export default Item;
