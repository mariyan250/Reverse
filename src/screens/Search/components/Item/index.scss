@import '../../../../index.scss';

// box shadow - make it reusable >>> GLOBAL LEVEL

.item {
  background: $background-white;
  box-shadow: 1px 1px 3px $silver;

  .img-container::before {
    display: 'block';
    content: '';
    padding-top: 100%;
  }

  img {
    object-fit: cover;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
  }

  h2 {
    font-size: 24px;
  }

  h3 {
    font-size: 16px;
  }

  h4 {
    font-size: 24px;
  }
}
