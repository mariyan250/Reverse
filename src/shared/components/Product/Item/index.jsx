// React and Style
import React from 'react';
import './index.scss';

// Router
import { Link } from 'react-router-dom';

// Translation
import { useTranslation } from 'react-i18next';

// TODO use translation for currency
function Item({ id, images, name, location, date, price }) {
  const { t } = useTranslation();

  return (
    <li className="item app-bg-white box-shadow">
      <Link to={`/item/${id}`} className="app-text-secondary">
        <section className="img-container d-flex overflow-hidden position-relative">
          <img
            src={images[0]}
            alt={t('search_screen.item.img.alt')}
            className="img-fluid position-absolute"
          />
        </section>

        <section className="p-3">
          <h6 className="my-3 text-center">{name}</h6>
          <p className="mt-4 mb-2 font-weight-light">
            <i className="fas fa-map-marker-alt mr-2" />
            {location}
          </p>
          <article className="d-flex justify-content-between align-items-end">
            <p>
              <i className="fas fa-clock mr-2" />
              {date}
            </p>
            <h4>{price} $</h4>
          </article>
        </section>
      </Link>
    </li>
  );
}

export default Item;
