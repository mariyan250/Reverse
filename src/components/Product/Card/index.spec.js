import React from 'react';
import { render } from 'tests/test-utils';
import Item from './index';

describe('Item', () => {
  it('should render the component', () => {
    const { container } = render(<Item images={[]} />);
    expect(container).toMatchSnapshot();
  });
});
