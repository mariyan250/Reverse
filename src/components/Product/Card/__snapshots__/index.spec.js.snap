// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render the component 1`] = `
<div>
  <li
    class="item app-bg-white overflow-hidden"
  >
    <a
      class="app-text-secondary"
      href="/item/undefined"
    >
      <div
        class="container d-flex position-relative justify-content-center"
      >
        <img
          alt="search_screen.item.img.alt"
          class="img img-fluid position-absolute p-5"
        />
      </div>
      <div
        class="py-3 px-4 h-100 app-bg-gray"
      >
        <p
          class="mb-4"
        />
        <p
          class="mb-3"
        >
          <i
            class="fas fa-map-marker-alt mr-2"
          />
        </p>
        <p
          class="mb-4"
        >
          <i
            class="fas fa-clock mr-2"
          />
        </p>
        <p>
           
          general.currency
        </p>
      </div>
    </a>
  </li>
</div>
`;
